src/modules/fancyzones/FancyZonesLib/MonitorWorkAreaMap.cpp

```diff
@@ -61,7 +61,27 @@ void MonitorWorkAreaMap::AddWorkArea(HMONITOR monitor, std::unique_ptr<WorkArea>
     m_workAreaMap.insert({ monitor, std::move(workArea) });
 }
 
+FancyZonesDataTypes::WorkAreaId MonitorWorkAreaMap::GetParent(HMONITOR monitor) const
+{
+    if (m_workAreaParents.contains(monitor))
+    {
+        return m_workAreaParents.at(monitor);
+    }
+    
+    return FancyZonesDataTypes::WorkAreaId{};
+}
+
 void MonitorWorkAreaMap::Clear() noexcept
 {
+    // save parent ids
+    m_workAreaParents.clear();
```

Comment:
change looks good, but I'd prefer this out of Clear() function as it's a bit odd having it there. Maybe  extract to separate function and call in `UpdateWorkAreas` (and if needed elsewhere)